libs/block-scripts/shared/generate-dist-example-graph.js

```diff
@@ -4,17 +4,65 @@ import fs from "fs-extra";
 
 import { blockDistDirPath, blockRootDirPath } from "./paths.js";
 
-export const generateDistExampleGraph = async () => {
-  const exampleGraphInDistFilePath = path.resolve(
+/* @todo - Where do we validate these metadata files, should it be here? */
+
+/**
+ * @param {Record<string, any>} assetsManifest
+ * @param {string} assetsManifestPath
+ */
+export const generateDistExampleGraph = async (
+  assetsManifest,
+  assetsManifestPath,
+) => {
+  const exampleGraphScriptName = assetsManifest["example-graph.cjs"];
+
+  const distExampleGraphPath = path.resolve(
     blockDistDirPath,
     "example-graph.json",
   );
-  const exampleGraphFilePath = path.resolve(
+
+  const sourcePlainTextExampleGraphPath = path.resolve(
     blockRootDirPath,
     "example-graph.json",
   );
+  const sourceScriptExampleGraphPath = exampleGraphScriptName
+    ? path.resolve(blockDistDirPath, exampleGraphScriptName)
+    : undefined;
+
+  if (
+    sourceScriptExampleGraphPath &&
+    (await fs.pathExists(sourceScriptExampleGraphPath))
+  ) {
+    if (await fs.pathExists(sourcePlainTextExampleGraphPath)) {
+      throw new Error(
+        "Both example-graph.js and example-graph.json exist. Please only supply one of them.",
+      );
+    }
+
+    const {
+      default: { default: exampleGraph },
+    } = await import(sourceScriptExampleGraphPath);
 
-  if (await fs.pathExists(exampleGraphFilePath)) {
-    await fs.copy(exampleGraphFilePath, exampleGraphInDistFilePath);
+    if (exampleGraph === undefined || !exampleGraph.entities) {
+      throw new Error(
+        "The default export of the example-graph script must be a valid example graph, containing a top-level 'entities' property.",
+      );
+    }
+    await fs.writeJson(distExampleGraphPath, exampleGraph);
+
+    await fs.remove(sourceScriptExampleGraphPath);
+    // eslint-disable-next-line no-param-reassign
+    delete assetsManifest["example-graph.cjs"];
+
+    await fs.writeJson(assetsManifestPath, assetsManifest, { spaces: 2 });
+    return;
+  } else if (await fs.pathExists(sourcePlainTextExampleGraphPath)) {
+    await fs.copy(sourcePlainTextExampleGraphPath, distExampleGraphPath);
+    return;
   }
+
+  /* @todo - Should this be optional? */
```

Comment:
I guess if a block doesn't need an entity then it shouldn't have to pass an `example-graph`? My concern is that if a user accidentally moves the script, or changes something in a way our complicated webpack and local tooling setup doesn't pick up, it would just silently ignore it rather than complaining.

It might be better to default to users having to explicitly provide a file, but just leaving it empty of entities / elements

Reply:
We could instead have a warning in the console

Reply:
People basically never see warnings in consoles, especially if `codegen` is just called as part of `yarn build`

Reply:
(Not a leading question), would you like it if we make it optional in the spec but required in `block-scripts`?

Reply:
I think that's fine. The template is opinionated about things that aren't required in the spec. Really I don't mind what we do except that if we are going to add a new requirement to the spec, I would like a justification for it being required.

Reply:
I've come full circle and I think it might be problematic if this is required and a block doesn't require an entity, or if it expresses its examples through `Variants`.

Going to make it optional for now and throw a warning as you recommended

Reply:
[Make the example-graph file optional](https://github.com/blockprotocol/blockprotocol/pull/1223/commits/858ae83f0f31420fad60ad094d8b0cf49b9d006a)